src/main.ts

```diff
@@ -97,7 +97,7 @@ export async function run(): Promise<void> {
       )
     } else {
       await exec.exec(
-        `git --git-dir ${repoDir}/.git --work-tree ${repoDir} checkout --progress --force ${commit}`
+        `git --git-dir ${repoDir}/.git --work-tree ${repoDir} checkout --progress --force ${checkoutInfo.ref}`
```

Comment:
Below is the implementation of getCheckoutInfo, where the ref is set when startPoint is not specified. If it is not set, an empty string is used, which should result in checking out the default branch, so I believe there should be no issues.

Could you please confirm if there are any overlooked considerations?

https://github.com/namespacelabs/nscloud-checkout-action/blob/6c7a74d91a4c53d13e8c759c05fcbd2d182f2d50/src/main.ts#L226-L253